manual_user_install.sh

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash
+
+# tested with a clean 64 bit installation of Raspbian-Lite
+
+function install_core (){
+    echo "Installing OVOS core"
+    echo
+    pip3 install -U pip
+
+    # install ovos-core - ovos-backend-client has conflicting dependencies
+    pip3 install git+https://github.com/OpenVoiceOS/ovos-backend-client
+    pip3 install git+https://github.com/OpenVoiceOS/ovos-core
+    pip3 install git+https://github.com/OpenVoiceOS/ovos-audio
+    pip3 install git+https://github.com/OpenVoiceOS/ovos-ocp-audio-plugin
+    pip3 install tornado
+    pip3 install git+https://github.com/OpenVoiceOS/ovos-messagebus
+    pip3 install adapt-parser
+    pip3 install padacioso
+
+    # dinkum listener
+    pip3 install git+https://github.com/OpenVoiceOS/ovos-dinkum-listener
+    pip3 install git+https://github.com/OpenVoiceOS/ovos-vad-plugin-webrtcvad
+    pip3 install git+https://github.com/OpenVoiceOS/ovos-vad-plugin-silero
```

Comment:
only need silero, not webrtcvad, its one or the other

Reply:
As long as silero installs, it works much better

Reply:
I think I pulled webrtcvad back in because of a dependency.  Let me remove everything we don't need and then I will test remove it again.

Reply:
dinkum-listener fails without webrtcvad. I will include webrtcvad for the time being.. Here are the related messages from ovos-systemd-dinkum-listener[647]
voice - ovos_dinkum_listener.service:__init__:123 - INFO - Starting Voice Service
voice - ovos_dinkum_listener.service:on_alive:58 - INFO - DinkumVoiceService is alive.
voice - ovos_dinkum_listener.service:_before_start:280 - INFO - Starting service...
voice - ovos_bus_client.session:reset_default_session:531 - INFO - New Default Session S>
voice - ovos_dinkum_listener.service:_connect_to_bus:379 - INFO - Connected to Mycroft C>
voice - ovos_plugin_manager.utils:load_plugin:146 - WARNING - Could not find the plugin >
voice - ovos_plugin_manager.vad:create:129 - ERROR - VAD plugin ovos-vad-plugin-webrtcva>
Traceback (most recent call last):
   File "/home/ovos/.local/lib/python3.9/site-packages/ovos_plugin_manager/vad.py", line 127, in create
     return clazz(plugin_config)
 TypeError: 'NoneType' object is not callable
 Traceback (most recent call last):
   File "/home/ovos/.local/bin/ovos-systemd-dinkum-listener", line 30, in <module>
     main(ready_hook=notify_ready, stopping_hook=notify_stopping)
   File "/home/ovos/.local/lib/python3.9/site-packages/ovos_dinkum_listener/__main__.py", line 24, in main
     service = OVOSDinkumVoiceService(on_ready=ready_hook, on_error=error_hook,
   File "/home/ovos/.local/lib/python3.9/site-packages/ovos_dinkum_listener/service.py", line 150, in __init__
     self.vad = OVOSVADFactory.create()
   File "/home/ovos/.local/lib/python3.9/site-packages/ovos_plugin_manager/vad.py", line 127, in create
     return clazz(plugin_config)
 TypeError: 'NoneType' object is not callable
ovos-dinkum-listener.service: Main process exited, code=exited, status=1/FAILURE
   Failed with result 'exit-code'.
Failed to start OVOS Dinkum Listener.

Reply:
When it fails, is silero installed and specified in mycroft.conf? Mine is running with only silero installed

Reply:
You have identified a problem.  Silero is installed and I'm using the default mycroft.conf.
        // vad module can be any plugin, by default it is not used
        // recommended plugin: "ovos-vad-plugin-silero"
        "module": "ovos-vad-plugin-webrtcvad",
        "ovos-vad-plugin-silero": {"threshold": 0.2},
        "ovos-vad-plugin-webrtcvad": {"vad_mode": 3}

Looks to me like the build completed correctly:
Successfully built ovos-vad-plugin-silero
Installing collected packages: mpmath, humanfriendly, sympy, protobuf, packaging, numpy, flatbuffers, coloredlogs, onnxruntime, ovos-vad-plugin-silero

I also looked in syslog as well as mycroft logs and there is NO MENTION OF SILERO.

Reply:
it needs a companion change in mycroft.conf, this script should probably copy mycroft.conf over from this repo (wget ? ) which is finetuned for a headless install

Reply:
[This is the mycroft.conf](https://github.com/OpenVoiceOS/raspbian-ovos/blob/headless/stage-config/01-mycroft-conf/files/mycroft.conf) that has `silero` enabled by default, but it may be too much for what is needed here.

or maybe have the script run a `ovos-config set` command.  I am not sure exactly how that works, but it might

Reply:
I'm partial to the edits being made right in the script because it teaches a new user exactly what to do. It also forces the changes to be visible.  Let me get to it and see if it's reasonable. Thank you!

Reply:
maybe add a function to check for an existing `~/.config/mycroft/mycroft.conf` file and uses `sed` to add or replace the `"VAD": {"module": <vad-plugin>}` to it if it exists, and just use a simple `echo` command to create the file and the config value needed.

Reply:
Perfect.  I get it. If we're talking about having this number of potential changes, the context given by the file is important and we'll just copy it onto the local path.  A series of "set"s would lose something.  Plus we can comment this file as required. I will test and if it works I will commit the new script tonight.

Reply:
it may have issues with some of the hotwords not loading because I had preloaded models so it could talk without being connected to the internet.  Mainly used for the wifi setup of first boot.  maybe just add a default `mycroft.conf` file somewhere that can be used.  Or better yet, place this script in a directory, and have all the files that are needed in that directory also.  You could put in the systemd files and a small mycroft.conf to be used by this script

Reply:
You have a very similar mycroft.conf in stage-ovos/02-ovos/files/mycroft.conf. My thought was to overwrite that file. It looked to me like you were hoping to have both the script and an image build working from the same code base, and I've been thinking of the problem from that perspective.  For the script build, I am just copying the appropriate files from the image build folder tree. It's all working right now (I commented out the piper references) and I'm going to do a clean build test in the morning. If no new issues by then, I'll push it for you to have a look.

Reply:
I'm thinking that how we handle customizations for the mycroft.conf file is a big question. I opened an issue to discuss a configurator enhancement.  This also relates to how we handle the padatious question.